### PR TITLE
UID and Label based uploads

### PR DESCRIPTION
**Pull request by rentzso**
works on #215

Still pending engine upload refactoring (partial work on [this branch](https://github.com/scitran/core/tree/engine-uploads)

Needs to sync with [scitran/reaper](https://github.com/scitran/reaper) before merge.

Also this will need some changes to the endpoint used for bootstrapping in fly/fly.
[This route](https://github.com/scitran/core/pull/232/files#diff-e0978ea022e5e946d30dfa3911881b5eR85) has been kept temporarily for retrocompatibility.

### CODE REVIEW

**Comment by nagem** [comment]
LGTM pending engine uploader changes, nice integ test addition!

File: api/placer.py [hunk 1]
```diff
@@ -146,8 +146,7 @@ def process_file_field(self, field, info):
         self.id             = container._id
         self.container      = container.container
 
-        for x in ('type', 'instrument', 'measurements', 'tags', 'metadata'):
-            info[x] = r_metadata.get(x) or info[x]
+        info.update(r_metadata)
```
Comment (nagem) [inline api/placer.py]: This is cool, I didn't know python dicts had this method

**Comment by kofalt** [comment]
Reviewing now - let's not block on the engine placer since this diff accomplishes quite a bit :)

**Comment by kofalt** [comment]
LGTM! Taking hierarchy changes on faith.

@rentzso Will there be a PR on scitran/reaper that will block this from landing?
My feature branch will need to PR after this one lands.

**Comment by rentzso** [comment]
@kofalt thanks!

I need to sync with @gsfr for changes on the reaper. I will keep you posted but you can probably work on the engine placer from this branch.

**Comment by kofalt** [comment]
Syncing as per team discussion :+1:

**Comment by gsfr** [comment]
Btw, LGTM, too. :wink:

**Pull request merged**